### DISCUSSION TITLE
Public API - Get nested objects

### DISCUSSION THREAD

**Discussion by reichert** [Budibase/budibase#6117] [Help] [answered]
Hi Guys,

Is it possible through the public API to retrieve nested objects (with relationships)?
- endpoint being used: https://budibase.internal.myserver.com/api/public/v1/tables/tableId/rows/rowId
- I'm using internal budibase tables

If so is it possible to specify the unitl what nesting leve it should return?
- as example, if nesting leve = 2 it would return a full structure until order_item given this tables structure:
- customer
  - orders
    - order_item
      - coupon_code

thanks in advance.

**Comment by melohagan** [comment, marked as answer]
Well it is a REST API, so to get a specific row, you'll need an id. You'll most likely need to do multiple calls to get the data you want.

What you're suggesting sounds like a GraphQL query to me, which we don't offer for the public API.

**Reply by reichert** [reply]
Thanks @melohagan - yes, we are going to through making multiple calls,
Thanks for your answer!

**Comment by maiducan** [comment]
This would be handy for me now. There is an overhead in creating multiple api calls instead of having an option to set number of nested levels and do just one.
